aws-lambda/src/databricks_cdk/resources/service_principals/service_principal_secrets.py

```diff
@@ -0,0 +1,156 @@
+import json
+import logging
+from typing import Optional
+
+import boto3
+from databricks.sdk import AccountClient
+from databricks.sdk.errors import NotFound
+from databricks.sdk.service.oauth2 import SecretInfo
+from pydantic import BaseModel
+
+from databricks_cdk.resources.service_principals.service_principal import get_service_principal
+from databricks_cdk.utils import CnfResponse, get_account_client
+
+SECRETS_MANAGER_RESOURCE_PREFIX = "/databricks/service_principal/secrets"
+
+logger = logging.getLogger(__name__)
+
+
+class ServicePrincipalSecretsCreationError(Exception):
+    pass
+
+
+class ServicePrincipalSecretsProperties(BaseModel):
+    service_principal_id: int
+
+
+class ServicePrincipalSecretsResponse(CnfResponse):
+    secrets_manager_arn: str
+    secrets_manager_name: str
+
+
+def create_or_update_service_principal_secrets(
+    properties: ServicePrincipalSecretsProperties, physical_resource_id: Optional[str] = None
+) -> ServicePrincipalSecretsResponse:
+    """
+    Create or update service principal secrets on databricks.
+    If service principal secrets already exist, it will return the existing service principal secrets.
+    If service principal secrets doesn't exist, it will create a new one.
+    """
+    account_client = get_account_client()
+
+    if physical_resource_id:
+        return update_service_principal_secrets(properties, physical_resource_id, account_client)
```

Comment:
in case have a use case and we want to recreate the secret again, will we have to delete it first?

Reply:
Thanks, there was a bug actually, because secrets manager name didn't take into account secret id. Fixed it, https://github.com/godatadriven/databricks-cdk/pull/1074/commits/c5dad5f0b7d58721b6b9bca13170cf42facbac7f

Secrets manager name looks a bit ugly though, but it works😅
<img width="696" alt="image" src="https://github.com/user-attachments/assets/ab8fa402-1e3d-42f3-a7bb-617b9e13cf95" />

Reply:
nice catch!